### ISSUE TITLE
hooks for gist

### ISSUE THREAD

**Issue by steveb85** [marshalhayes/gitbucket-githooks-plugin#10]
this plugin works great in our workflow for our gits, but I was wondering if there is a way to have it work for gists as well? In the file system the gists do have a hooks folder, but i don’t think this plugin supports them?

**Comment by marshalhayes** [comment]
I'm actually not sure about this. I guess you're wanting to run a hook when a gist is updated/created?

**Comment by steveb85** [comment]
yes exactly @marshalhayes . in the .gitbucket/gist folder each gist does have a hooks folder, so it’s potentially possible, just wondering if it’s a simple implementation or not.

**Comment by marshalhayes** [comment]
I'm looking at the [gist plugin](https://github.com/gitbucket/gitbucket-gist-plugin) now and it looks like this might be doable.

On the gist controller, there are two actions that are interesting:
- a post to create a new gist ([see here](https://github.com/gitbucket/gitbucket-gist-plugin/blob/9a81964e44f1903ee2b48f563fbf3489e783c2cf/src/main/scala/gitbucket/gist/controller/GistController.scala#L100))
- a post to update an existing gist ([see here](https://github.com/gitbucket/gitbucket-gist-plugin/blob/9a81964e44f1903ee2b48f563fbf3489e783c2cf/src/main/scala/gitbucket/gist/controller/GistController.scala#L138))

They both ultimately end up committing any changes. 

I think the real question to ask here is whether or not, when the Gist plugin commits these changes, does GitBucket call the `receiveHooks` defined in [Plugin.scala](https://github.com/marshalhayes/gitbucket-githooks-plugin/blob/f73304db813f0ff80c4c680d77cfa4e0b0eb0374/src/main/scala/Plugin.scala#L17)? 

Out of curiosity, have you tried setting up a post-receive hook for a gist already? It might just work.

**Comment by steveb85** [comment]
@marshalhayes i did try it when I discovered the hooks folder within the gists, but didn’t get anything, I didn’t go too deep because I assumed it didn’t work as no one had mentioned it, I can do some more thorough testing if you like. 

I think the updating is the only thing you ant to worry about, when you create a gist you can’t have a hook, because it doesn’t exist u til it’s saved, so perhaps it could be simple, but also require input from the gist plugin? 

Thanks for checking it out!
